Cheatsheet.md

```diff
@@ -0,0 +1,504 @@
+const str = "GeeksforGeeks      ";
+const st = str.trim();
+console.log(st);
+```
+*Output:*
+   
+    GeeksForGeeks
+
+### [5. charAt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/charAt)
+
+**Description:**
+Returns the character at the specified index in a string.
+
+**Syntax:**
+ * _string.charAt(index)_
+
+**Return value:**
+String, representing the character at the specified index, or an empty string if the index number is not found
+**Example:**
+```
+const str = 'JavaScript is object oriented language';
+console.log(str.charAt(9));
+```  
+  *Output:*
+
+    t
+    
+## Many forms of for loops
+
+### [1. for...in](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in)
+
+**Description:**
+ Loops through the properties of an object. The block of code inside the loop will be executed once for each property.
+
+**Syntax:**
+ * _for (var in object) {<br>
+  code block to be executed<br>
+ }_
+
+| Parameter | Description |
+| :---      | :---        |
+|var	|Required. A variable that iterates over the properties of an object|
+|object	| Required. The specified object that will be iterated|
+
+**Example:**
+ ```  
+ onst array1 = ['a', 'b', 'c'];
+ for (const element of array1) {
+ 	console.log(element);
+ }
+```
+*Output:*
+
+    "a"
+    "b"
+    "c"
+
+### [2.for...of](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Statements/for...of)
+
+**Description:**
+Loops through the values of an iterable objects.
+
+**Syntax:**
+ * _for (variable of iterable) {<br>
+  // code block to be executed<br>
+}_
+
+| Parameter | Description |
+| :---      | :---        |
+| variable | For every iteration the value of the next property is assigned to the variable. Variable can be declared with const, let, or var.|
+|iterable | An object that has iterable properties.|
+
+**Example:**
+```  
+const languages = { first : "C", second : "Java", 
+			third : "Python", fourth : "PHP", 
+		        	fifth : "JavaScript" }; 
+// iterate through every property of the 
+// object languages and print all of them 
+// using for..in loops 
+for (itr in languages) 
+{ 
+	console.log(languages[itr]); 
+} 
+```
+*Output:* 
+    
+    C
+    Java
+    Python
+    PHP
+    JavaScript
+
+### [3. for loop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Loops_and_iteration#for_statement)
+
+**Description:**
+Loops through a block of code a number of times based on whether a given condition testes to ve true or false.
+
+**Syntax:**
+ * _for (statement 1; statement 2; statement 3) {<br>
+  // code block to be executed<br>
+}_
+
+| Parameter | Description |
+| :---      | :---        |
+| _statement 1_ : variable declaration| Executed (one time) before the execution of the code block.|
+| _statement 2_ : test condition| Defines the condition for executing the code block. |
+| _statement 3_ : change in variable| Executed (every time) after the code block has been executed. |

```

Comment:
Remove all this and replace with a student-friendly description: something like "traditional index-based loop as found in many languages"

Reply:
Replace this description or all the descriptions?

Reply:
@vegetabill ? ^